src/algebra/divisibility/basic.lean

```diff
@@ -91,14 +91,20 @@ end semigroup
 
 section monoid
 
-variables [monoid α]
+variables [monoid α] {a b : α}
 
 @[refl, simp] theorem dvd_refl (a : α) : a ∣ a := dvd.intro 1 (mul_one a)
 theorem dvd_rfl : ∀ {a : α}, a ∣ a := dvd_refl
 instance : is_refl α (∣) := ⟨dvd_refl⟩
 
 theorem one_dvd (a : α) : 1 ∣ a := dvd.intro a (one_mul a)
 
+lemma dvd_of_eq (h : a = b) : a ∣ b := by rw h
+lemma dvd_of_eq' (h : a = b) : b ∣ a := by rw h
+
+alias dvd_of_eq ← eq.dvd
+alias dvd_of_eq' ← eq.dvd'
```

Comment:
Is there precedent for this? I'd expect to be forced to write `symm` to flip the equality.

Reply:
There's `eq.le`/`eq.ge`, or `has_lt.lt.ne`/`has_lt.lt.ne'`.

Reply:
Those don't feel the same though; they're workarounds for the fact that `ge` and `le` are mathematically distinct notions that happen to be merged in Lean (`has_lt.lt.ne'` is pretending to be `ge.ne`). I don't think there is any precedent for a backwards `dvd` though.

Reply:
Is there any downside to having `dvd_of_eq'`, though?

Reply:
Having aliases for every symmetric version of a lemma can encourage you to be sloppy and not keep things a consistent way around throughout a proof.

Reply:
Hmm... not having aliases for every symmetric version *forces* you to be sloppy and not keep things a consistent way around when you need to prove two lemmas that are the same up to swapping sides. This is in fact why I want to introduce that lemma.

And surely being able to be sloppy is in general a *good* thing, right?

Reply:
I agree with Eric, it seems overkill to have a symmetric version `eq.rel'` for every `rel` that is reflexive. Moreover, I would find it much clearer to have `a_eq_b.symm.dvd` than `a_eq_b.dvd'`, since the `'` is very unspecific.

Reply:
Yes, but in that case `dvd` forms a preorder (or even a partial order when there's only one unit), so it's conceptually very close to `≤`.

Reply:
Now removed, but I am sad.